### PR TITLE
Delete bot embeds by reacting to it

### LINKED ISSUE #12: Enable deleting mentions by reacting with an emoji to it
It would be nice if the bot would remove its message if it is reacted to with a certain (if possible, configurable) emoji. If possible, this should only be possible for the person who initially invoked the mention.

### PR DESCRIPTION
**Pull request by chandler05**
## Purpose
Close #12 
## Approach
Use the 🗑️ reaction on bot embed mentions that you triggered to delete them

### CODE REVIEW

File: src/events/mention/MentionDeleteEventHandler.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export default class MentionDeleteEventHandler implements EventHandler<'messageReactionAdd'> {
+    public readonly eventName = 'messageReactionAdd';
+
+    private logger = log4js.getLogger( 'MentionDeleteEventHandler' );
+
+    public onEvent = async ( { message }: MessageReaction, user: User ): Promise<void> => {
+        this.logger.info( `User ${ user.tag } is attempting to delete message '${ message.id }'` );
+
+	const embeds = message.embeds;
```
Comment (SPGoding) [inline src/events/mention/MentionDeleteEventHandler.ts:14]: Something's not right with indent here ;D

Not sure why it wasn't caught by eslint.

File: src/events/mention/MentionDeleteEventHandler.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        const mentionEmbed = embeds[0];
+        const footer: string = mentionEmbed.footer.text;
```
Comment (SPGoding) [inline src/events/mention/MentionDeleteEventHandler.ts:16]: I guess there might be errors if the message doesn't contain an embed, or if the embed doesn't have a footer.

File: src/events/mention/MentionDeleteEventHandler.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        const userTag = footer.match( /.{3,32}#[0-9]{4}/ )[1];
```
Comment (SPGoding) [inline src/events/mention/MentionDeleteEventHandler.ts:17]:
```suggestion
        const userTag = footer.match( /.{3,32}#[0-9]{4}/ )[0];
```

File: src/events/mention/MentionDeleteEventHandler.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        if ( user.tag === userTag ) {
+            try {
+                await message.delete;
```
Comment (SPGoding) [inline src/events/mention/MentionDeleteEventHandler.ts:21]:
```suggestion
                await message.delete();
```

**Comment by chandler05** [comment]
wastebasket

**Review by violine1101 — CHANGES_REQUESTED** [review]
Looks good so far! Only issue I have is that the bot could theoretically also delete messages with this that it didn't send itself. There should be a check that asserts that the author is MojiraBot.

File: src/events/reaction/ReactionAddEventHandler.ts [hunk 5]
```diff
@@ -48,6 +50,9 @@ export default class ReactionAddEventHandler implements DiscordEventHandler<'mes
 		} else if ( BotConfig.request.logChannel.includes( messageReaction.message.channel.id ) ) {
 			// Handle reopening a user request
 			return this.requestReopenEventHandler.onEvent( messageReaction, user );
+		} else if ( messageReaction.message.author.id === this.botUserId && messageReaction.emoji.name === BotConfig.embedDeletionEmoji ) {
```
Comment (chandler05) [inline src/events/reaction/ReactionAddEventHandler.ts:53]: @violine1101 there is! Thanks for reviewing!

Comment (violine1101) [reply src/events/reaction/ReactionAddEventHandler.ts:53]: Ah, I see, thanks. I would have expected it inside the actual event, but that's fine too.

**Review by violine1101 — APPROVED** [review]

**Pull request merged**